File: core/feature04b.js
function feature04b() {
  var parameters = extractParameters(["M"]);
  return feature04b_core(parameters);
}

function feature04b_core(parameters) {
  if (parameters == null) {
    return null;
  }
  var M = getParameter(parameters, "M");
  var result = {};
  var Rad = Math.PI / 180.0;
  var C = 400.0;
  var N = M - C;
  var B = Math.acos(N / C) / Rad;
  var W = 8.0 * Math.sin(B * Rad);
  var A = Math.acos((N + W) / C) / Rad;
  result["A"] = 2 * A;
  result["B"] = 90 - 2 * A;
  return result;
}

File: core/feature04b.go
package main

import (
	"math"
)

func feature04B(mm float64) (aa, bb float64) {
	Rad := math.Pi / 180.0
	cc := 400.0
	nn := mm - cc
	bb = math.Acos(nn/cc) / Rad
	ww := 8.0 * math.Sin(bb*Rad)
	aa = math.Acos((nn+ww)/cc) / Rad
	aa = 2 * aa
	bb = 90 - aa
	return
}
